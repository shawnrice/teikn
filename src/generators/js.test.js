const startCase = require('lodash/startCase');
const { generator } = require('./generator-js');
const Generator = require('./JS');
const pkg = require('../../package.json');

describe('es5 generator tests', () => {
  const tokens = [{ name: 'test', type: 'color', value: '#CCCCCC' }];

  test('it generates things', () => {
    const output = `module.exports = {
  /**
   *  Type: color
   */
  test: '#CCCCCC'
}`;
    expect(generator(tokens)).toBe(output);
  });
});

describe('es5 tests', () => {
  const tokens = [{ name: 'test', type: 'color', value: '#CCCCCC' }];

  test('it generates things', () => {
    const output = `/**
 * ${startCase(pkg.name)} v${pkg.version}
 * Generated null
 *
 * This file is generated and should be commited to source control
 *
 */
module.exports = {
  /**
   *  Type: color
   */
  test: '#CCCCCC'
}`;
    expect(new Generator({ dateFn: () => 'null' }).generate(tokens)).toBe(output);
  });
});
