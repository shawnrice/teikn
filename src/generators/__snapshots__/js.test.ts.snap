// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`es5 tests it generates things 1`] = `
"/**
 * Teikn v0.1.5
 * Generated null
 *
 * This file is generated and should be commited to source control
 *
 */
module.exports = {
  /**
   *  Use for body fonts
   *  Type: font-family
   */
  body: '\\"Roboto Condensed\\", Arial, sans',
  /**
   *  the primary branding color
   *  Type: color
   */
  primary: 'aliceblue'
}"
`;
